pmd-core/src/main/java/net/sourceforge/pmd/PMDConfiguration.java

```diff
@@ -493,75 +516,162 @@ public void setMinimumPriority(RulePriority minimumPriority) {
      */
     @Deprecated
     public void setInputPaths(String inputPaths) {
-        List<String> paths = new ArrayList<>();
-        Collections.addAll(paths, inputPaths.split(","));
+        List<Path> paths = new ArrayList<>();
+        for (String s : inputPaths.split(",")) {
+            paths.add(Paths.get(s));
+        }
         this.inputPaths = paths;
     }
 
     /**
      * Set the input paths to the given list of paths.
      * @throws NullPointerException If the parameter is null
+     * @deprecated Use {@link #setInputPathList(List)}
      */
+    @Deprecated
```

Comment:
Ideally, if we merge #4161 first into master(pmd6), then master->pmd/7.0.x and update this PR, these changes should disappear.

And then we can think about, removing the deprecated methods already (if we ourselves don't use them in pmd7 anymore).

Reply:
👍